src/test/test_prob_distr.c

```diff
@@ -0,0 +1,1197 @@
+		ok = false;						      \
+	}								      \
+} while (0)
+
+/**
+ * Test the logit and logistic functions.  Confirm that they agree with
+ * the cdf, sf, icdf, and isf of the standard Logistic distribution.
+ * Confirm that the sampler for the standard logistic distribution maps
+ * [0, 1] into the right subinterval for the inverse transform, for
+ * this implementation.
+ */
+static void
+test_logit_logistic(void *arg)
+{
+  (void) arg;
+
+	static const struct {
+		double x;	/* x = logit(p) */
+		double p;	/* p = logistic(x) */
+		double phalf;	/* p - 1/2 = logistic(x) - 1/2 */
+	} cases[] = {
+		{ -HUGE_VAL, 0, -0.5 },
+		{ -1000, 0, -0.5 },
+		{ -710, 4.47628622567513e-309, -0.5 },
+		{ -708, 3.307553003638408e-308, -0.5 },
+		{ -2, .11920292202211755, -.3807970779778824 },
+		{ -1.0000001, .2689414017088022, -.23105859829119776 },
+		{ -1, .2689414213699951, -.23105857863000487 },
+		{ -0.9999999, .26894144103118883, -.2310585589688111 },
+		{ -4.000000000537333e-5, .49999, -1.0000000000010001e-5 },
+		{ -4.000000000533334e-5, .49999, -.00001 },
+		{ -4.000000108916878e-9, .499999999, -1.0000000272292198e-9 },
+		{ -4e-9, .499999999, -1e-9 },
+		{ -4e-16, .5, -1e-16 },
+		{ -4e-300, .5, -1e-300 },
+		{ 0, .5, 0 },
+		{ 4e-300, .5, 1e-300 },
+		{ 4e-16, .5, 1e-16 },
+		{ 3.999999886872274e-9, .500000001, 9.999999717180685e-10 },
+		{ 4e-9, .500000001, 1e-9 },
+		{ 4.0000000005333336e-5, .50001, .00001 },
+		{ 8.000042667076272e-3, .502, .002 },
+		{ 0.9999999, .7310585589688111, .2310585589688111 },
+		{ 1, .7310585786300049, .23105857863000487 },
+		{ 1.0000001, .7310585982911977, .23105859829119774 },
+		{ 2, .8807970779778823, .3807970779778824 },
+		{ 708, 1, .5 },
+		{ 710, 1, .5 },
+		{ 1000, 1, .5 },
+		{ HUGE_VAL, 1, .5 },
+	};
+	double relerr_bound = 3e-15; /* >10eps */
+	size_t i;
+	bool ok = true;
+
+	for (i = 0; i < arraycount(cases); i++) {
+		double x = cases[i].x;
+		double p = cases[i].p;
+		double phalf = cases[i].phalf;
+
+		/*
+		 * cdf is logistic, icdf is logit, and symmetry for
+		 * sf/isf.
+		 */
+		CHECK_RELERR(logistic(x), cdf_logistic(x, 0, 1));
+		CHECK_RELERR(logistic(-x), sf_logistic(x, 0, 1));
+		CHECK_RELERR(logit(p), icdf_logistic(p, 0, 1));
+		CHECK_RELERR(-logit(p), isf_logistic(p, 0, 1));
+
+		CHECK_RELERR(cdf_logistic(x, 0, 1), cdf_logistic(x*2, 0, 2));
+		CHECK_RELERR(sf_logistic(x, 0, 1), sf_logistic(x*2, 0, 2));
+		CHECK_RELERR(icdf_logistic(p, 0, 1), icdf_logistic(p, 0, 2)/2);
+		CHECK_RELERR(isf_logistic(p, 0, 1), isf_logistic(p, 0, 2)/2);
+
+		CHECK_RELERR(cdf_logistic(x, 0, 1), cdf_logistic(x/2, 0, .5));
+		CHECK_RELERR(sf_logistic(x, 0, 1), sf_logistic(x/2, 0, .5));
+		CHECK_RELERR(icdf_logistic(p, 0, 1), icdf_logistic(p, 0,.5)*2);
+		CHECK_RELERR(isf_logistic(p, 0, 1), isf_logistic(p, 0, .5)*2);
+
+		/*
+		 * For p near 0 and p near 1/2, the arithmetic of
+		 * translating by 1 loses precision.
+		 */
+		if (fabs(p) > DBL_EPSILON && fabs(p) < 0.4) {
+			CHECK_RELERR(cdf_logistic(x, 0, 1),
```

Comment:
Hmm why did we check the value of `p` if we are not using it at all it in the two checks below?

If you meant to use `p` instead of `x` in: `CHECK_RELERR(cdf_logistic(x, 0, 1), cdf_logistic(x*2 + 1, 1, 2));`
what does the value of `p` matter anyhow? Isn't this formula as true as `CHECK_RELERR(cdf_logistic(x, 0, 1), cdf_logistic(x*2, 0, 2));` for any `p`?